.github/scripts/validate_config_sync.py

```
@@ -0,0 +1,81 @@
+import os
+import sys
+import json
+import re
+from typing import Set
+
+def extract_env_vars_from_code(file_path: str) -> Set[str]:
+    """Extracts environment variable names from os.getenv() calls in a Python file."""
+    try:
+        with open(file_path, 'r', encoding='utf-8') as f:
```

Ruff (0.14.10)

Unnecessary mode argument

Remove mode argument

(UP015)